### PR TITLE
Feature/add multicontext inference processing

### PR DESCRIPTION
**Pull request by dchvs99**

### CODE REVIEW

File: rr/ai/ai_manager.py [hunk 1]
```diff
@@ -190,9 +190,7 @@ def __init__(
         self.on_new_prediction_cb_ = None
 
     def install_callback(self, on_new_prediction_cb_):
-        self._mutex.acquire()
         self.on_new_prediction_cb_ = on_new_prediction_cb_
-        self._mutex.release()
```
Comment (emmadrigal) [inline rr/ai/ai_manager.py:195]: Why was this removed?

File: rr/gstreamer/media_manager.py [hunk 2]
```diff
@@ -122,7 +122,7 @@ def stop_media(self):
     def install_callback(self, callback):
         for key in self._Dict:
```
Comment (emmadrigal) [inline rr/gstreamer/media_manager.py]: You can iterate over the key and the value simultaneously
```for key, value in _Dict.items():```

File: rr/smart_cctv.py [hunk 3]
```diff
@@ -82,15 +86,19 @@ def _create_ai_manager(self, config):
     def __init__(self, config):
         # Make sure the AI manager is the first class to be created, otherwise
```
Comment (emmadrigal) [inline rr/smart_cctv.py]: `the AI managers are the first classes`

**Review by emmadrigal — CHANGES_REQUESTED** [review]

File: rr/ai/ai_manager.py (hunk 1 above)
Comment (dchvs99) [reply rr/ai/ai_manager.py:195]: We are handling this in parallel this time!

File: rr/ai/ai_manager.py [hunk 4]
```diff
@@ -190,9 +190,7 @@ def __init__(
         self.on_new_prediction_cb_ = None
 
     def install_callback(self, on_new_prediction_cb_):
-        self._mutex.acquire()
```
Comment (michaelgruner) [inline rr/ai/ai_manager.py:193]: If you're doing this, don't you need to remove the mutex?

File: rr/stream/stream_manager.py [hunk 5]
```diff
@@ -58,16 +58,24 @@ def __init__(
...
+        for key in ai_manager_dict:
+            cb.update({key: OnNewImage(
+                ai_manager_dict[key],
+                model,
+                disp_width,
+                disp_height)})
+
+            self.ai_manager_dict[key].install_callback(cb_prediction)
+
+        self.media_manager.install_callback(cb)
```
Comment (michaelgruner) [inline rr/stream/stream_manager.py]: Install callbacks

**Review by michaelgruner — CHANGES_REQUESTED** [review]

**Review by michaelgruner — APPROVED** [review]

**Review by emmadrigal — APPROVED** [review]
